frontend/src/store/modules/subscription.ts

```diff
@@ -0,0 +1,86 @@
+import axios from "axios";
+import {
+  Subscription,
+  FeatureType,
+  PlanType,
+  SubscriptionState,
+} from "../../types";
+import { FEATURE_MATRIX } from "./plan";
+
+const state: () => SubscriptionState = () => ({
+  subscription: undefined,
+});
+
+const getters = {
+  subscription: (state: SubscriptionState) => (): Subscription | undefined => {
+    return state.subscription;
+  },
+
+  currentPlan: (state: SubscriptionState) => (): PlanType => {
+    // TODO: this is used for align with current logic - TEAM plan is default plan
+    return state.subscription?.plan ?? PlanType.TEAM;
+  },
+
+  feature:
```

Comment:
What's the relation between this and the one in plan.ts

Reply:
It will replace plan.ts later, we will use the subscription store instead of plan. Will totally replace and remove the plan.ts in following PR.